### ISSUE TITLE
Enforcing collections be non-null

### ISSUE THREAD

**Issue by efenderbosch** [checkstyle/checkstyle#6810]
I'd like to enforce that collections in our domain objects be non-null.

```
public class Order {
    // if this isn't initialized, flag it
    private List<LineItem> lineItems = new ArrayList<>();

    // if this isn't tagged w/ @Nonnull, flag it
    @Nonnull
    public List<LineItem> getLineItems() {
       return lineItems;
    }

    // if this isn't tagged w/ @Nonnull, flag it
   public void setLineItems(@Nonnull List<LineItems> lineItems) {
        this.lineItems = lineItems;
    }
}
```

Is this something that would make a good checkstyl check? Or be more appropriate for spotbugs/fb-contrib?

I'm willing to do the work to write the detector, but not sure if this is something that either project would like contributed, or would be more appropriate for our internal checks.

**Comment by romani** [comment]
@efenderbosch , it is good to read about Checkstyle limitations - https://checkstyle.org/writingchecks.html#Limitations

Checkstyle can not recognize hierarchy of types, so we need to be exact of scope of validation.
Checkstyle can not get details from another file, so if your domains make hierarchy with fancy overrides, it might be problem, so we will have to skip such cases to not produce false positives.

You need to provide more examples of usages and imagine that Check is already implemented so please provide imaginary config and code it violates and not-violates.

Good example of how to propose new Check - https://github.com/checkstyle/checkstyle/issues/5899

if there might be false-positives but you still want to have in Checkstyle, we can host it at https://github.com/sevntu-checkstyle/sevntu.checkstyle it is our experimental sand box project.

**Comment by romani** [comment]
No details provided on possible solution, to my mind it is not doable by Checkstyle.
If new ideas come up we can reopen issue
